### ISSUE TITLE
Exclamation mark not working from Magic Keyboard to Windows 10

### ISSUE THREAD

**Issue by sgrienen** [debauchee/barrier#754]
Server: macOS High Sierra, latest available patches applied
Barrier version 2.3.2-snapshot-210c2b70

Client: Windows 10 1809
Barrier version 2.3.2-snapshot-210c2b70

### Steps to reproduce bug ###

I'm using a Swiss German QWERTZ Apple Magic keyboard on my server, and when I try to type an exclamation mark, it just doesn't do anything on Windows 10. The exclamation mark is just beside the Enter key, above the " ¨ ", so it has to be reached using Shift. It's the only character I haven't been able to type in Windows from that keyboard, even if the mapping is sometimes tricky.

**Comment by p12tic** [comment]
Let's not close valid bugs without a valid reason.

**Comment by mumrich** [comment]
I can confirm that the issue still exists, even on a Windows-11 (Server) -> Windows-11 (Client) scenario.
I'm also using a Swiss-German keyboard-layout.
Using Barrier:
![image](https://user-images.githubusercontent.com/10463536/195395903-b20b337a-6915-4a14-aea2-b8b13f14fd7b.png)

I can neither input `¨` nor `!` on a client machine, which is quite annoying for a programmer... 😅:
- Cannot use `ctrl` + `¨` for opening terminals
- cannot negate a Boolean operation due to `!` missing.

The key 00_1B does not seem to work at all:
![image](https://user-images.githubusercontent.com/10463536/195393652-94d08b35-8d3a-47bb-a2c8-746e5176bec4.png)

I've tried to press the 00_1B key with [sharpkeys](https://apps.microsoft.com/store/detail/sharpkeys/XPFFCG7M673D4F):
Server:
![image](https://user-images.githubusercontent.com/10463536/195394063-6ee3ad99-3779-4167-8edc-4b152972a718.png)

Client:
![image](https://user-images.githubusercontent.com/10463536/195394759-89ddc1b0-c5a7-40ce-85a8-b4ff30132ed8.png)

**Comment by lumisoft** [comment]
Having the same issue using barrier 2.4.0 as server on Debian 12.2 and client on Windows 10 with Swiss french keyboard layout.  "!" always missing.

**Comment by lucamiani** [comment]
I have the same issue on 2.4 on Windows 10 to 10

**Comment by Dewege** [comment]
Same issue here
Version 2.4.0-release-3e0d758b
server: macos ventura 13.6.7 swiss german keyboard (magic keyboard without num-block)
to client: W10 swiss german keyboard

### Test 1 (change keyboard layout)
I tried to additionally configure a US international keyboard the windows-settings and switched to that, which barrier seems to respect for certain keys, but not for others:
-   ! should now be available via shift and „1“ (as far as I know and as the screen keyboard shows), but that results in a + (as on the swiss german keyboard layout)
-   AltGr and „2“ results in superscript 2 (correct for US-kbd, would be @ on swiss german kbd)
 
### Test 2 (use keystroke() to capture ! but produce another char on client)
I also tried to configure hotkeys via GUI and tested if I can produce other characters (e.g. an „a“) on W10 client when pressing ! on the Mac side (needs shift, as can be seen in screenshot above of @mumrich). 
It shows as `keystroke(Shift+!)` with action `keystroke(a,W10PC)` 

**That works** but with a caveat: now _everytime_ when I want to type „!" on the server, an „a“ is _always_ appearing on the client (even when the mouse is on the server (mac))
This happens also if I edit and use a barrier.conf file with the following line in section: options
`keystroke(Shift+Exclaim) = keystroke(a,W10PC)`

### maybe related:
I also can not write > or < to the client unless I specify this as hotkeys in GUI
  `keystroke(<)` with action `keystroke(<,W10PC)`
  `keystroke(Shift+>)` with action `keystroke(Shift+>,W10PC)`
and again: now they appear _always_ on the client only, I can not type them locally on the server anymore.
Do I use that in a wrong way?

### TL;DR summary
- can not find a way to produce „!“ character on windows client from mac server
- a soon as keystroke() rules are defined those key combinations can not be typed on server anymore, but appear _always_ only on client
